app/src/main/java/org/mozilla/fenix/components/metrics/AppLaunchTimeMeasurement.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.os.Process
+import android.os.SystemClock
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.perf.Stat
+
+/**
+ * Handles the logic of figuring out launch time for cold, warm and hot startup.
+ */
+class AppLaunchTimeMeasurement(private val stats: Stat = Stat()) {
+
+    // when user launched the time. i.e press app_icon, launched a link.
+    private var processLaunchTimeStampNanoTime: Long? = null
+    // we are considering screen to be visible after the first pre draw call.
+    private var elapsedNanoTimestampOnHomeActivityPreDraw: Long? = null
+
+    fun onApplicationOnCreate() {
+        // There is disk read involve and we dont want to block the main thread.
+        GlobalScope.launch(Dispatchers.IO) {
```

Comment:
I think avoiding to use `GlobalScope` is ideal. The coroutine could hang if ever something happens with `stats` or reading from the file. Maybe using `MainScope()` ?

Reply:
Hmm, good point. I guess I always associated the main scope with dealing with stuff related to UI.  Also, I thought Mainscope had same lifetime as GlobalScope? Maybe keep track of the job and if its not done when needed, cancel it?

Reply:
Ah yes it is. but you can still launch form a different dispatchers IIRC so `MainScope().launch(Dispatchers.IO)` but preferably would be to have our own scope 😄